### PR TITLE
chore(EMS-3559): iterations - SonarCloud intentionality errors

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR fixes Sonarcloud intentionality errors

## Resolution :heavy_check_mark:
* Various changes to codebase for intentionality

### CODE REVIEW

File: src/ui/server/helpers/string/index.test.ts [hunk 1]
```diff
@@ -100,15 +100,15 @@ describe('server/helpers/string', () => {
...
+        const result = stringsAreDefined('test');
 
         expect(result).toEqual(false);
       });
     });
 
     describe('both strings are undefined', () => {
       it('should return false', () => {
-        const result = stringsAreDefined(undefined, undefined);
+        const result = stringsAreDefined();
```
Comment (ttbarnes) [inline src/ui/server/helpers/string/index.test.ts:111]: Seems that we can remove the test on line 49 with this change

File: src/ui/types/index.d.ts [hunk 2]
```diff
@@ -85,6 +85,10 @@ import {
 import { InsuranceFeedbackVariables } from './feedback';
 import { SanitiseValueObjParams } from './sanitise-value';
 
+type FieldsWithValues = {
+  [key: string]: string;
+};
```
Comment (ttbarnes) [inline src/ui/types/index.d.ts:90]: Could we use `ObjectType` instead? Few instances of `ObjectType` declared, will be a nice and quick DRY change 😎

File: src/ui/server/helpers/string/index.test.ts (hunk 1 above)
Comment (Zainzzkk) [reply src/ui/server/helpers/string/index.test.ts:111]: that is a different function?

File: src/ui/types/index.d.ts (hunk 2 above)
Comment (ttbarnes) [reply src/ui/types/index.d.ts:90]: Can we move this into its own file like the other types?

Happy to do after this is merged

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**